src/main/java/com/teragrep/lsh_01/config/ConversionFactory.java

```diff
@@ -0,0 +1,142 @@
+/*
+  logstash-http-input to syslog bridge
+  Copyright 2024 Suomen Kanuuna Oy
+
+  Derivative Work of Elasticsearch
+  Copyright 2012-2015 Elasticsearch
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+package com.teragrep.lsh_01.config;
+
+import com.teragrep.lsh_01.authentication.BasicAuthentication;
+import com.teragrep.lsh_01.conversion.IMessageHandler;
+import com.teragrep.lsh_01.conversion.JsonConversion;
+import com.teragrep.lsh_01.conversion.RegexConversion;
+import com.teragrep.lsh_01.conversion.RelpConversion;
+import com.teragrep.lsh_01.pool.IManagedRelpConnection;
+import com.teragrep.lsh_01.pool.Pool;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+public final class ConversionFactory {
+
+    private final static Logger LOGGER = LogManager.getLogger(ConversionFactory.class);
+
+    private final String splitType;
+    private final String regexPattern;
+    private final Pool<IManagedRelpConnection> pool;
+    private final SecurityConfig securityConfig;
+    private final BasicAuthentication basicAuthentication;
+    private final LookupConfig lookupConfig;
+
+    public ConversionFactory(
+            Map<String, String> configuration,
+            Pool<IManagedRelpConnection> pool,
+            SecurityConfig securityConfig,
+            BasicAuthentication basicAuthentication,
+            LookupConfig lookupConfig
+    ) {
+        // if system property is not specified, defaults to config file (the Map)
+        this(
```

Comment:
this System.getProperty allows the object to be runtime configurable at any point in time when calling for a new suchObject

Reply:
Took the System.getProperty calls out and changed the tests to use properties from a resource file instead so they work without this.
Currently the default configs are in a file /etc/config.properties. I like the idea of having the defaults as a decorator in the Map, but it would require to refactor all config objects now and in my opinion could be done in another PR. There's an issue open (#78) for that. Configuration for json and regex splitting are now working as intended.